### DISCUSSION TITLE
Is a Set() call synchronous or asynchronous?

### DISCUSSION THREAD

**Discussion by waltforme** [redis/go-redis#1890] [General]
For example, in the sample code in project's README.md, we have
```
err := rdb.Set(ctx, "key", "value", 0).Err()
if err != nil {
        panic(err)
}
```
Looks like `Set` is synchronous because `err` must be assigned with a value for the next line of code to run.

Is there an asynchronous way to `Set`? I'm asking because I am working with a use case where
1) I'll be frequently setting a value;
2) my redis server is on cloud, where there is a considerable delay which may conflict with the high frequency.

**Comment by vmihailenco** [comment]
All go-redis API is synchronous. You can use channels/goroutines if you need async execution.

**Reply by waltforme** [reply]
I see. Thanks for the reply!
